### PR TITLE
Migrate to python 3.8

### PR DESCRIPTION
**Pull request by pvizeli**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -345,7 +345,7 @@ jobs:
     needs: prepare
     strategy:
       matrix:
-        python-version: [3.7, 3.8]
+        python-version: [3.8]
```
Comment (pvizeli) [inline .github/workflows/ci.yaml:348]: We only run on the version which we use inside our container, so we don't need check multible version

Comment (frenck) [reply .github/workflows/ci.yaml:348]: Agree, however, the tests don't pass on Python 3.8.

I don't know or understand why at this point.

Comment (pvizeli) [reply .github/workflows/ci.yaml:348]: ok

**Comment by balloob** [comment]
It's because `mock_coro` is not awaited, which fails the tests. Using the built-in `patch` will work.

**Comment by balloob** [comment]
Pushed fix

**Comment by frenck** [comment]
Looks good to me now 🎉 

One small thing, the 3.7 can be removed from the first job step matrix.

If we leave the matrix in, we can easily add a new Python version (before migration) in the future.

**Pull request merged**